lib/internal/histogram.js

```diff
@@ -53,6 +53,30 @@ function isHistogram(object) {
   return object?.[kHandle] !== undefined;
 }
 
+function* infinite() {
```

Comment:
I don't think this is a good implementation.

Reply:
I agree!

Reply:
It will be much simple to just return new timerified function. 

Do we really need cache? Is there any real use case of it? Note, built-in `Function.prototype.bind` never cache, and I also don't know Web platform API do similar cache.

Reply:
I read the upstream PR. The intention is to improve performance. So if we remove the cache feature, the will be a significant performance regression, also, this must be a BREAKING change

Reply:
Upstream: https://github.com/nodejs/node/pull/37136

Reply:
Quote:

> Old: perf_hooks/usertiming.js n=100000: 92,378.01249733355
> New: perf_hooks/usertiming.js n=100000: 270,393.5280638482